src/main/java/edu/harvard/iq/dataverse/util/DataSourceProducer.java

```diff
@@ -0,0 +1,70 @@
+    //
+    // HINT: PGSimpleDataSource would work too, but as we use a connection pool, go with a javax.sql.ConnectionPoolDataSource
+    // HINT: PGXADataSource is unnecessary (no distributed transactions used) and breaks ingest.
+    className = "org.postgresql.ds.PGConnectionPoolDataSource",
+    
+    user = "${MPCONFIG=dataverse.db.user}",
+    password = "${MPCONFIG=dataverse.db.password}",
+    url = "jdbc:postgresql://${MPCONFIG=dataverse.db.host}:${MPCONFIG=dataverse.db.port}/${MPCONFIG=dataverse.db.name}",
+    
+    // If we ever need to change these pool settings, we need to remove this class and create the resource
+    // from web.xml. We can use MicroProfile Config in there for these values, impossible to do in the annotation.
+    //
+    // See also https://blog.payara.fish/an-intro-to-connection-pools-in-payara-server-5
+    // Payara DataSourceDefinitionDeployer default value = 8
+    minPoolSize = 10,
+    
+    // HINT: Payara DataSourceDefinitionDeployer default value = 32
+    // HINT: Harvard Dataverse is fine for a while with 64
+    maxPoolSize = 100,
+    
+    // "The number of seconds that a physical connection should remain unused in the pool before the connection is closed for a connection pool. "
+    // Payara DataSourceDefinitionDeployer default value = 300 (seconds)
+    maxIdleTime = 300)
+    
+    // These are documented at https://docs.payara.fish/community/docs/5.2020.6/documentation/payara-server/jdbc/advanced-connection-pool-properties.html#full-list-of-properties
+    //
+    // All these vars cannot be configured via MPCONFIG as Payara doesn't support this (yet). To be enhanced.
+    // See also https://github.com/payara/Payara/issues/5024
+    /*
+    properties = {
+        "fish.payara.pool-resize-quantity=${MPCONFIG=dataverse.db.pool-resize-quantity}",
+        "fish.payara.is-connection-validation-required=${MPCONFIG=dataverse.db.is-connection-validation-required}",
+        "fish.payara.connection-validation-method=${MPCONFIG=dataverse.db.connection-validation-method}",
+        "fish.payara.validation-table-name=${MPCONFIG=dataverse.db.validation-table-name}",
+        "fish.payara.validation-classname=${MPCONFIG=dataverse.db.validation-classname}",
+        "fish.payara.connection-leak-timeout-in-seconds=${MPCONFIG=dataverse.db.connection-leak-timeout-in-seconds}",
+        "fish.payara.connection-leak-reclaim=${MPCONFIG=dataverse.db.connection-leak-reclaim}",
+        "fish.payara.statement-timeout-in-seconds=${MPCONFIG=dataverse.db.statement-timeout-in-seconds}",
+        "fish.payara.statement-leak-timeout-in-seconds=${MPCONFIG=dataverse.db.statement-leak-timeout-in-seconds}",
+        "fish.payara.statement-leak-reclaim=${MPCONFIG=dataverse.db.statement-leak-reclaim}",
+        "fish.payara.slow-query-threshold-in-seconds=${MPCONFIG=dataverse.db.slow-query-threshold-in-seconds}",
+        "fish.payara.log-jdbc-calls=${MPCONFIG=dataverse.db.log-jdbc-calls}"
```

Comment:
I find it confusing to see all the MPCONFIG values here when they don't work yet. (Thanks for opening https://github.com/payara/Payara/issues/5024 to address this.)

In practice, I'm not sure how many of these we (Harvard Dataverse) use but perhaps this list could be simplified down to the two mentioned in https://github.com/IQSS/dataverse/pull/7048#issuecomment-656264879 which are:

```
properties = {
    "fish.payara.is-connection-validation-required=true",
    "fish.payara.validation-table-name=setting"
}
```

Once there is a list with hard coded values like this, it's more obvious that uncommenting them will (presumably) have an effect.

Reply:
Removed in c583909.